src/screen.c

```diff
@@ -4875,6 +4878,46 @@ set_chars_option(win_T *wp, char_u *value, int is_listchars, int apply,
 		    break;
 		}
 
+		if (!is_listchars && STRCMP(tab[i].name.string, "ellipsis") == 0)
+		{
+		    if (round == 0)
+		    {
+			ellipsis_len = 0;
+			while (*s != NUL && *s != ',')
+			{
+			    c1 = get_encoded_char_adv(&s);
+			    if (char2cells(c1) > 1)
+				return field_value_err(errbuf, errbuflen,
+				     e_wrong_character_width_for_field_str,
+				     tab[i].name.string);
```

Comment:
This check may be unnecessary if the popupmenu.c code can handle double-width chars in "ellipsis".

Reply:
okay got it

Reply:
Oh wait, unprintable chars still should be allowed. So the condition should be changed to `!vim_isprintc(c1)` instead.